cranelift/codegen/src/binemit/mod.rs

```diff
@@ -73,6 +73,9 @@ pub enum Reloc {
     /// jalr ra, ra, 0
     RiscvCall,
 
+    /// Riscv64 TLS GD (https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/riscv-elf.adoc#global-dynamic)
+    RiscvTlsGd,
```

Comment:
Support for this needs to be implemented in cranelift_object. Should be pretty similar to https://github.com/bytecodealliance/wasmtime/pull/4995/files#diff-b825ad4c01be6ae3d42dd6ace012f2815e5e52458abc4f436526a5c4aaa769adR711

Reply:
@bjorn3  Do you mean the name `Reloc::RiscvCall ` should be `Reloc::R_RISCV_CALL`????

Reply:
This is what I meant:

```diff
diff --git a/cranelift/object/src/backend.rs b/cranelift/object/src/backend.rs
index f40a75822..c06c2e3e4 100644
--- a/cranelift/object/src/backend.rs
+++ b/cranelift/object/src/backend.rs
@@ -720,6 +720,32 @@ impl ObjectModule {
                     0,
                 )
             }
+            Reloc::RiscvTlsGdHi20 => {
+                assert_eq!(
+                    self.object.format(),
+                    object::BinaryFormat::Elf,
+                    "RiscvTlsGdHi20 is not supported for this file format"
+                );
+                (
+                    RelocationKind::Elf(object::elf::R_RISCV_TLS_GD_HI20),
+                    RelocationEncoding::Generic,
+                    0,
+                )
+
+            }
+            Reloc::RiscvPCRelLo12I => {
+                assert_eq!(
+                    self.object.format(),
+                    object::BinaryFormat::Elf,
+                    "RiscvPCRelLo12I is not supported for this file format"
+                );
+                (
+                    RelocationKind::Elf(object::elf::R_RISCV_PCREL_LO12_I),
+                    RelocationEncoding::Generic,
+                    0,
+                )
+
+            }
             // FIXME
             reloc => unimplemented!("{:?}", reloc),
         };
```

Reply:
ok.